### PR TITLE
Improve serialization of joinGame

### PR DESCRIPTION
**Pull request by Kwasow**
The `joinGame` function now uses a JsonSerializer object to parse the received json, instead of doing it manually. I also improved the `removePlayer` function's readability.

### CODE REVIEW

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 1]
```diff
@@ -37,7 +35,7 @@ object GameState {
     private var nextId = 0
 
     // Constants
-    private const val BASE_URL = "http://158.101.160.143:42069"
+    private const val BASE_URL = "http://10.0.2.2:42069"
```
Comment (sormys) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt]: wrong url

Comment (Kwasow) [reply app/src/main/java/com/pokerio/app/utils/GameState.kt]: Damn it, we need to do something about this

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 2]
```diff
@@ -303,38 +305,41 @@ object GameState {
...
+            return
+        }
+
+        val removedPlayer = players.find { it.playerID == playerHash }
+            ?: throw Exception("Player to be removed not found")
 
-            players.removeIf { it.playerID == playerHash }
-            playerRemovedCallbacks.forEach { it.value(removedPlayer) }
+        if (removedPlayer.isAdmin && newAdmin == null) {
+            throw Exception("Admin removed without new admin given")
```
Comment (mikolaj-pirog) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt]:
```suggestion
            throw Exception("Admin removed, but new admin was not set")
```

Comment (mikolaj-pirog) [reply app/src/main/java/com/pokerio/app/utils/GameState.kt]: Seems like a better wording

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 3]
```diff
@@ -303,38 +305,41 @@ object GameState {
...
+            ?: throw Exception("Player to be removed not found")
 
-            players.removeIf { it.playerID == playerHash }
-            playerRemovedCallbacks.forEach { it.value(removedPlayer) }
+        if (removedPlayer.isAdmin && newAdmin == null) {
+            throw Exception("Admin removed, but new admin was not set")
+        }
+
+        if (removedPlayer.isAdmin) {
+            val isThisPlayerNewAdmin = players.find { sha256(it.playerID) == newAdmin } != null
```
Comment (sormys) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt]: You are doing the same find here and in line 330, can't you find the player and simply check if  he is not null in `if`?

**Review by sormys — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/pokerio/app/utils/GameState.kt (hunk 3 above)
Comment (Kwasow) [reply app/src/main/java/com/pokerio/app/utils/GameState.kt]: I removed this as well as improving the actual removal of the player (line 336)

**Review by sormys — APPROVED** [review]

**Review by mikolaj-pirog — APPROVED** [review]

**Pull request merged**